lib/langs.rb

```diff
@@ -28,4 +34,93 @@ class Language
   extend LanguageHelpers
 end
 
+module LanguageVersionHelpers
+
+  private
+
+  def canonical_version(version)
+    version.match(/^\./) ? "0#{version}" : version
+  end
+
+  # Translate version requirement syntax to a String or Array of Strings that Gem::Dependency.new understands
+  # Pipfile and Cargo.toml files support * and ^ (wildcard and caret) requirements, which Ruby does not
+  # See https://www.python.org/dev/peps/pep-0440/#version-matching
+  # And: https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html
+  def translate_requirement(req, language)
+    if language == :python
```

Comment:
Python and Rust translation both require partially the same treatment (because they both have wildcards) and partially different (different syntax for `~>` requirements, only Rust has caret requirements). Does it make sense to handle them together in this method?

Reply:
Not a big fan of checking for languages within one method. Much prefer a straightforward substitution chain (e.g., `req.sub('~=', '~>').sub('==', '=')`) in each language definition.

Reply:
Ok, but the code for wildecards will be identical. Turn that into a common method?

Reply:
Yes.